lib/matplotlib/figure.py

```diff
@@ -325,6 +325,10 @@ def __init__(self,
             frameon = rcParams['figure.frameon']
 
         self.bbox_inches = Bbox.from_bounds(0, 0, *figsize)
+        if not all(np.isfinite(_) for _ in figsize):
```

Comment:
You can just write `np.isfinite(figsize).all()` (and similarly below): the ufunc will cast the input to an ndarray.